### ISSUE TITLE
[enhancement] Support deployment with Istio

### ISSUE THREAD

**Issue by FZambia** [centrifugal/helm-charts#23]
Looks like we have a problem with port names which prevent Istio to properly pass WS traffic: 

https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/

We can:

* Have a better port names
* support setting `appProtocol` field (should not break things for k8s < 1.18 )

**Comment by LvLs9** [comment]
I've tried another setup and looks like it works with current chart. Maybe I'm was wrong with something... 

It seems that the first time istio was not able to set up the protocol on its own, no ideas why.

**Comment by LvLs9** [comment]
Hm, actually not at all.
Setup on production from scratch caught same problem. Looks like really will be good to hardcode protocols for Istio

**Comment by LvLs9** [comment]
Finally it connects, but after 10 seconds. And before connection it doing some strange things... 
Istio logs on second screenshot

![Снимок экрана 2021-11-28 в 7 54 47 PM](https://user-images.githubusercontent.com/34302242/143775610-33e7c0bf-a33a-4899-a51d-4bdc4d103fb9.png)
![Снимок экрана 2021-11-28 в 7 53 02 PM](https://user-images.githubusercontent.com/34302242/143775552-bf44a7f9-435f-483e-b154-2258f54cc013.png)

**Comment by LvLs9** [comment]
Yes! Strange, but... I've setup appProtocol in my custom service - and it connects without any issues. 

So, maybe we can just add appProtocol/setup name fields - so, Istio will strictly use correct protocol. Latest Istio 1.12

**Comment by FZambia** [comment]
@LvLs9 I found several issues in Istio which look very similar to this. For example this one - https://github.com/istio/istio/issues/32127. An this - https://github.com/istio/istio/issues/29427.

Unfortunately these issues are closed without any resolution from Istio team. So I think adding a possibility to define `appProtocol` for each port as first step here makes sense. I think should be empty by default and set in spec only if value provided. Thus we won't use it in older k8s. What do you think?

**Comment by LvLs9** [comment]
@FZambia Yes, agreed 💪 

I can try build MR with fix

**Comment by LvLs9** [comment]
@FZambia Before merging PR...

I've tried to delete istiod pod (which will reconfigure network) - and it's worked correctly. But not sure it's really good solution.

**Comment by FZambia** [comment]
So we now have a possibility to define appProtocol for each service port (added in https://github.com/centrifugal/helm-charts/pull/24) - thus we have an explicit way to tell Istio protocols to use.

For the future I think would be awesome to investigate whether we should change current port names in some way to inherit automatic Istio deployment without customizing options.

**Comment by LvLs9** [comment]
Should we close that?

**Comment by FZambia** [comment]
@LvLs9 Let's leave this open for now – hope in the future we will figure out the remaining parts. I'd like Centrifugo work with Istio out-of-the-box without appProtocol configuration – but requires some research and local experiments – I am not fully understand what Istio actually does.

**Comment by FZambia** [comment]
#156 solves this
